### PR TITLE
Interval

### PR DESCRIPTION
**Pull request by eric-feng-2011**
Added Interval expression and tests

- Modified Literal to parse CalendarInterval days
- Added date_add function
- Added dateAddInterval function
- Included Scala and Spark SQL tests

### CODE REVIEW

**Review by wzheng — COMMENTED** [review]
Just one question!

File: src/enclave/Enclave/ExpressionEvaluation.h [hunk 1]
```diff
@@ -3,8 +3,11 @@
 #include <functional>
 #include <typeinfo>
 #include <cmath>
+// #include <ctime>
```
Comment (chester-leung) [inline src/enclave/Enclave/ExpressionEvaluation.h:6]: Can you remove this comment as well?

File: src/enclave/Enclave/ExpressionEvaluation.h [hunk 2]
```diff
@@ -3,8 +3,11 @@
...
 #include <limits>
 
+#include <iostream>
```
Comment (chester-leung) [inline src/enclave/Enclave/ExpressionEvaluation.h:9]: Can you remove this additional (unnecessary?) include?

File: src/enclave/Enclave/ExpressionEvaluation.h [hunk 3]
```diff
@@ -265,6 +268,25 @@ class FlatbuffersExpressionEvaluator {
 
     case tuix::ExprUnion_Literal:
     {
+      auto * literal = static_cast<const tuix::Literal *>(expr->expr());
+      const tuix::Field *value = literal->value();
+
+      // If type is CalendarInterval, manually return a calendar interval field. 
+      // Otherwise 'days' disappears in conversion.
+      if (value->value_type() == 10) {
```
[the remark this answers is not in the document]
Comment (eric-feng-2011) [reply src/enclave/Enclave/ExpressionEvaluation.h:276]: The 10 is coming from 'target/scala-2.12/src_managed/flatbuffers/gen-cpp/Rows_generated.h' where it maps the FieldUnion_CalendarInterval type to 10. I'm using this to check for type matching.

File: src/enclave/Enclave/ExpressionEvaluation.h (hunk 1 above)
Comment (eric-feng-2011) [reply src/enclave/Enclave/ExpressionEvaluation.h:6]: Done!

File: src/enclave/Enclave/ExpressionEvaluation.h (hunk 2 above)
Comment (eric-feng-2011) [reply src/enclave/Enclave/ExpressionEvaluation.h:9]: done!

File: src/enclave/Enclave/ExpressionEvaluation.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (wzheng) [reply src/enclave/Enclave/ExpressionEvaluation.h:273]: Can you use `value->value_type() == tuix::FieldUnion_CalendarIntervalField` instead?

[the remark this answers is not in the document]
Comment (eric-feng-2011) [reply src/enclave/Enclave/ExpressionEvaluation.h]: Changed

**Review by wzheng — APPROVED** [review]
Thanks!

**Pull request merged**